Source/OIDIDToken.m

```diff
@@ -0,0 +1,152 @@
+/*! @file OIDIDToken.m
+    @brief AppAuth iOS SDK
+    @copyright
+        Copyright 2017 Google Inc. All Rights Reserved.
+    @copydetails
+        Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+ */
+
+#import "OIDIDToken.h"
+
+/*! Field keys associated with an ID Token. */
+static NSString *const kIssKey = @"iss";
+static NSString *const kSubKey = @"sub";
+static NSString *const kAudKey = @"aud";
+static NSString *const kExpKey = @"exp";
+static NSString *const kIatKey = @"iat";
+static NSString *const kNonceKey = @"nonce";
+
+#import "OIDFieldMapping.h"
+
+@implementation OIDIDToken
+
+@synthesize header = _header;
+@synthesize claims = _claims;
+@synthesize issuer = _issuer;
+@synthesize subject = _subject;
+@synthesize audience = _audience;
+@synthesize expiresAt = _expiresAt;
+@synthesize issuedAt = _issuedAt;
+@synthesize nonce = _nonce;
+
+- (instancetype)initWithIDTokenString:(NSString *)idToken {
+  self = [super init];
+  NSArray *sections = [idToken componentsSeparatedByString:@"."];
+  if (sections.count > 1) {
```

Comment:
I did a double-take here at the end of the method. Might be nice to:

```
// The header and claims are required.
if (sections.count <= 1) {
  return nil;
}
```

Reply:
Done.